host/sgx/enclavemanager.c

```diff
@@ -173,14 +173,20 @@ oe_enclave_t* oe_query_enclave_instance(void* tcs)
         OE_LIST_FOREACH(tmp, &oe_enclave_list_head, next_entry)
         {
             oe_enclave_t* enclave = tmp->enclave;
+
+            oe_mutex_lock(&enclave->lock);
+
             for (uint32_t i = 0; i < OE_COUNTOF(enclave->bindings); i++)
             {
                 if (enclave->bindings[i].tcs == (uint64_t)tcs)
                 {
                     ret = enclave;
+                    oe_mutex_unlock(&enclave->lock);
```

Comment:
Consider `break` for ending the loop instead for a single lock-unlock pair instead of an early return.